source/server/listener_manager_impl.cc

```diff
@@ -180,20 +205,38 @@ ListenerImpl::ListenerImpl(const envoy::api::v2::Listener& config, const std::st
   if (config.has_freebind()) {
     addListenSocketOptions(Network::SocketOptionFactory::buildIpFreebindOptions());
   }
-  if (config.has_tcp_fast_open_queue_length()) {
-    addListenSocketOptions(Network::SocketOptionFactory::buildTcpFastOpenOptions(
-        config.tcp_fast_open_queue_length().value()));
-  }
-
   if (!config.socket_options().empty()) {
     addListenSocketOptions(
         Network::SocketOptionFactory::buildLiteralOptions(config.socket_options()));
   }
 
   if (!config.listener_filters().empty()) {
-    listener_filter_factories_ =
-        parent_.factory_.createListenerFilterFactoryList(config.listener_filters(), *this);
+    switch (socket_type_) {
+    case Network::Address::SocketType::Datagram:
+      udp_listener_filter_factories_ =
+          parent_.factory_.createUdpListenerFilterFactoryList(config.listener_filters(), *this);
+      // Intentional return since udp filters do not need other configuration
+      return;
```

Comment:
Since you already removed filter chain from config in test, does it has to early return here?

Reply:
it need not, but as discussed in the other thread, will keep this as it is now